packages/console/src/pages/Profile/containers/VerificationCodeModal/index.tsx

```diff
@@ -73,12 +75,40 @@ const VerificationCodeModal = () => {
     } catch (error: unknown) {
       if (error instanceof HTTPError) {
         const logtoError = await error.response.json<RequestErrorBody>();
-        setError(logtoError.message);
-      } else {
-        setError(String(error));
+
+        // The error message below will be displayed as inline error message.
+        if (
+          logtoError.code === 'verification_code.code_mismatch' ||
+          logtoError.code === 'verification_code.exceed_max_try' ||
+          logtoError.code === 'verification_code.not_found'
+        ) {
+          setError(logtoError.message);
+
+          return;
+        }
+
+        // Other verification code errors will be displayed in a popup modal.
+        if (logtoError.code.startsWith('verification_code.')) {
+          await showModal({
+            ModalContent: logtoError.message,
+            type: 'alert',
+            cancelButtonText: 'general.got_it',
+          });
+
+          onClose();
+
+          return;
+        }
+
+        if (error.response.status !== 401) {
+          toast.error(logtoError.message);
+
+          return;
+        }
```

Comment:
nit: this logic is similar to `ChangePasswordModal`, maybe wrap into a funciton?

Reply:
Wrapped in a util function now, please take a look